lib/interfaces/include/EthernetInterface.h

```diff
@@ -0,0 +1,30 @@
+#ifndef __ETHERNETINTERFACE_H__
+#define __ETHERNETINTERFACE_H__
+
+//WHAT ETHERERNET INTERFACE NEEDS TO DO
+//1. get the protobuf message 2. map the values to a struct 3. send out those values to main charge sys
+
+//1. take CCUOutputValues struct and make into protobuf message 2. send the values out to ams
+
+struct CCUInputValues //I think this is correct but i dont know if i need to account for all the values in the protobuf message struct
+{
+    float maxCellTemp; //in degress C
+    float maxCellV; //in Volts
+};
+
+struct CCUOutputValues
+//Output values struct (what CCU is sending out)
+{
+    bool currentFlow;
+    float current_setpoint;
+};
+
+class EthernetInterface{
+public:
+    EthernetInterface();
+    CCUInputValues UpdatingValues();
+
+private:
+    CCUInputValues inputValues;
+};
+#endif
```

Comment:
I'd rewrite your ethernet interface to follow a similar paradigm as VCR. You have your real struct that you care about, your protoc struct, and your incoming bitstream. You can look at ProtobufMsgInterface.h in shared_firmware_interfaces to parse the bitstream into protoc struct, then your  ethernet interface should just be taking protoc struct and mapping those values to a struct you care about.